src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -270,6 +270,9 @@ public function get($id, $invalidBehavior = self::EXCEPTION_ON_INVALID_REFERENCE
 
                     $alternatives = array();
                     foreach ($this->services as $key => $associatedService) {
+                        if (isset($this->privates[$key])) {
```

Comment:
Makes sense for having `getServiceIds` only expose public id's.

Not sure why we want `$private = true`. Opposed to refactoring stuff relying on the fact it returns private ids.